### PR TITLE
Feature/int tests totals check

### PR DESCRIPTION
**Pull request by VaclavK**

### CODE REVIEW

File: .github/workflows/shared-app-integration-tests-workflow.yaml [hunk 1]
```diff
@@ -84,5 +86,5 @@ jobs:
             }
 
       - name: run integration tests
-        working-directory: ./src/${{ inputs.applicationFolder }}.Tests
-        run: dotnet test --filter TestCategory=IntegrationTest
\ No newline at end of file
+        working-directory: ./mammon/src/${{ inputs.applicationFolder }}.Tests
+        run: dotnet test --filter TestCategory=IntegrationTest --logger "console;verbosity=normal"
```
[the remark this answers is not in the document]
Comment (VaclavK) [reply .github/workflows/shared-app-integration-tests-workflow.yaml:90]: I will keep it there as it seems to improve logging

File: src/Mammon/Actors/SQLPoolActor.cs [hunk 2]
```diff
@@ -4,39 +4,60 @@ public class SQLPoolActor(ActorHost actorHost, CostCentreRuleEngine costCentreRu
...
+				{
+					var cost = new ResourceCost((decimal)(db.DTUAverage / totalDTU) * totalCost.Cost, totalCost.Currency);
+					allocatedCost += cost.Cost;
 
-				await ActorProxy.DefaultProxyFactory.CallActorWithNoTimeout<ICostCentreActor>(CostCentreActor.GetActorId(reportId, costCentre), nameof(CostCentreActor), async (p) => await p.AddCostAsync(resourceId, cost));
-			}
+					ResourceIdentifier dbRID = new(db.ResourceId);
+
+					var costCentre = costCentreRuleEngine.GetCostCentreForSQLDatabase(dbRID.Name);
+					if (!nsMetrics.TryGetValue(costCentre, out NSSQLCost? value))
```
[the remark this answers is not in the document]
Comment (VaclavK) [reply src/Mammon/Actors/SQLPoolActor.cs:40]: yeah.. in other places we only handle the negative one so I kept it for consistency but I will reverse here.. I hear you

[the remark this answers is not in the document]
Comment (VaclavK) [reply src/Mammon/Actors/SQLPoolActor.cs]: done now

File: src/Mammon/Actors/LAWorkspaceActor.cs [hunk 3]
```diff
@@ -1,54 +1,66 @@
...
-		ArgumentException.ThrowIfNullOrWhiteSpace(resourceId);
-		ArgumentException.ThrowIfNullOrWhiteSpace(reportId);
-		ArgumentNullException.ThrowIfNull(data);
-		ArgumentNullException.ThrowIfNull(totalCost);
+		ArgumentNullException.ThrowIfNull(request);
+
+		var totalCost = request.Resource.Cost;
+		var resourceId = request.Resource.ResourceId;
+		var tags = request.Resource.Tags;
+		var reportId = request.ReportRequest.ReportId;
```
Comment (rmjoia) [inline src/Mammon/Actors/LAWorkspaceActor.cs:16]: I think you could do something like 
`var (totalCost, resourceId, reportId) = request.ReportRequest;` or something.. I might be wrong.

https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/functional/deconstruct

Comment (VaclavK) [reply src/Mammon/Actors/LAWorkspaceActor.cs:16]: I don't think this applies here.. request and its members are not tuples

I do tuple deconstruction at other structures where I actually return tuple

**Review by rmjoia — APPROVED** [review]

**Pull request merged**